dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6724,6 +6725,21 @@ int processFrom(bool& isUnion, SELECT_LEX& select_lex, gp_walk_info& gwi, SCSEP&
   // Existed pushdown handlers won't get in this scope
   // MDEV-25080 Union pushdown would enter this scope
   // is_unit_op() give a segv for derived_handler's SELECT_LEX
+
+  // check INTERSECT or EXCEPT, that are not implemented
+  auto* nextSelect = select_lex.master_unit()->first_select()->next_select();
+  if (nextSelect)
+  {
+    if (nextSelect->get_linkage() == INTERSECT_TYPE)
+    {
+      setError(gwi.thd, ER_INTERNAL_ERROR, "INTERSECT is not supported by Columnstore engine", gwi);
+    }
+    else if (nextSelect->get_linkage() == EXCEPT_TYPE)
+    {
+      setError(gwi.thd, ER_INTERNAL_ERROR, "EXCEPT is not supported by Columnstore engine", gwi);
+    }
+  }
+
```

Comment:
Don't you need to quit when if there is an error @mariadb-LeonidFedorov ?

Reply:
++